storm-client/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -865,7 +879,7 @@ public static void setupDefaultUncaughtExceptionHandler() {
         Thread.setDefaultUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
                 public void uncaughtException(Thread thread, Throwable thrown) {
                     try {
-                        handleUncaughtException(thrown);
+                        handleUncaughtExceptionWithoutKillingProcess(thrown);
```

Comment:
Same question, if a thread gets an exception and the process is not shut down, will the thread be replaced?